ArcadiaParties-SPA/src/app/department/department-selector/department-selector.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnDestroy } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { Router } from '@angular/router';
+import { Observable, Subscription } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+
+import { loadDepartmentsAction } from '../actions/actions';
+import { selectDepartments, selectCurrentDepartment } from '../selector/selector';
+import { Department } from '../models/Department';
+import { DepartmentState } from '../reducers/reducer';
+
+@Component({
+  selector: 'app-department-selector',
+  templateUrl: './department-selector.component.html',
+  styleUrls: ['./department-selector.component.scss']
+})
+export class DepartmentSelectorComponent implements OnDestroy {
+  private currentDepartmentSubscription: Subscription;
+
+  departments$: Observable<Department[]> = this.store.pipe(select(selectDepartments));
+
+  selectedDepartmentId: number;
+
+  constructor(private store: Store<DepartmentState>, private router: Router) {
+    store.dispatch(loadDepartmentsAction());
+
+    this.currentDepartmentSubscription = this.store.pipe(select(selectCurrentDepartment), debounceTime(0))
+      .subscribe(id => {
+        this.selectedDepartmentId = id;
+        console.log(id);
+      });
+  }
+
+  onChange() {
+    this.router.navigate(['/home', this.selectedDepartmentId]);
```

Comment:
Could you rename route from `home` to `calendar`?

Reply:
Sure, done